packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -3274,6 +3299,51 @@ class SemanticsOwner extends ChangeNotifier {
 
   /// Update the semantics using [onSemanticsUpdate].
   void sendSemanticsUpdate() {
+    // Once the tree is up-to-date, verify that there's no invisible nodes in it.
+    assert(() {
+      final List<SemanticsNode> invisibleNodes = <SemanticsNode>[];
+      bool findInvisibleNodes(SemanticsNode rootNode) {
+        if (rootNode.rect.isEmpty) {
+          invisibleNodes.add(rootNode);
+        } else if (!rootNode.mergeAllDescendantsIntoThisNode) {
+          rootNode.visitChildren(findInvisibleNodes);
+        }
+        return true;
+      }
+
+      // The root node is allowed to be invisible when it has no children.
+      final SemanticsNode? rootSemanticsNode = this.rootSemanticsNode;
+      if (rootSemanticsNode != null) {
+        if (!rootSemanticsNode.mergeAllDescendantsIntoThisNode) {
+          rootSemanticsNode.visitChildren(findInvisibleNodes);
+        } else if (rootSemanticsNode.childrenCount > 0 && rootSemanticsNode.rect.isEmpty) {
+          invisibleNodes.add(rootSemanticsNode);
+        }
```

Comment:
Shouldn't the rootSemanticsNode also be considered invisible if it doesn't merge all descendants into it, has a childrenCount > 0, and an empty rect?

In other words, in the if branch, why is the rootNode not considered invisible if it has children and and empty rect?